tables_io/io_layer.py

```diff
@@ -358,3 +461,187 @@ def readPqToDataframes(basepath, keys=None, **kwargs):
         except Exception:  #pragma: no cover
             pass
     return dataframes
+
+
+def readPqToArrayDict(infile):
+    """ Open a parquet file and return a dictionary of `numpy.array`
+    """
+    df = readPqToDataframe(infile)
+    return dataframeToArrayDict(df)
+
+
+def readH5ToArrayDict(infile):
+    """ Open an h5 file and and return a dictionary of `numpy.array`
+    """
+    df = readHdf5ToDataframe(infile)
+    return dataframeToArrayDict(df)
+
+
+def readHdf5ToArrayDict(infile, groupname='None'):
+    """ Read in h5py hdf5 data, return a dictionary of all of the keys
+    """
+    hg, infp = readHdf5Group(infile, groupname)
+    data = hdf5GroupToArrayDict(hg)
+    infp.close()
+    return data
+
+
+def readFileToArrayDict(filename, fmt=None, groupname='None'):
+    """ Load training data
+
+    Parameters
+    ----------
+    filename : `str`
+        File to load
+    fmt : `str`
+        File format
+    groupname : `str` or `None`
+        For hdf5 files, the groupname for the data
+
+    Returns
+    -------
+    data : `dict` ( `str` -> `numpy.array` )
+        The data
+    """
+    if fmt is None:
+        _, ext = os.path.splitext(filename)
+        fmt = ext[1:]
+
+    fmtlist = ['hdf5', 'parquet', 'h5']
+    if fmt not in fmtlist:
+        raise NotImplementedError(f"File format {fmt} not implemented")
+    if fmt == 'hdf5':
+        data = readHdf5ToArrayDict(filename, groupname)
+    if fmt == 'parquet':
+        data = readPqToArrayDict(filename)
+    if fmt == 'h5':
+        data = readH5ToArrayDict(filename)
+    return data
+
+
+def getGroupInputDataSize(hg):
```

Comment:
Useful to distinguish size and len here! For 2D arrays (e.g. PDFs) you're returning len not size.  Which is fine, but useful to document carefully.